war/src/main/java/io/lumeer/engine/controller/configuration/ConfigurationManipulator.java

```diff
@@ -65,251 +61,136 @@
     *       the name of key located in 'config' field
-    * @return Object value of the given key
+    * @return Config object value of the given key
     */
-   public Object getConfiguration(final String collectionName, final String nameValue, final String key) {
-      return readValueFromDb(collectionName, nameValue, key).orElse(null);
+   public Config getConfiguration(final String collectionName, final String identificator, final String key) {
+      DataDocument document = systemDataStorage.readDocumentIncludeAttrs(collectionName, entryFilter(identificator, key),
+            Collections.singletonList(dataStorageDialect.concatFields(LumeerConst.Configuration.CONFIGS, "$")));
+      // we got only one subdocument otherwise there was null
+      return document != null ? new Config(document.getArrayList(LumeerConst.Configuration.CONFIGS, DataDocument.class).get(0)) : null;
    }
 
    /**
-    * Returns an Optional DataDocument representing configuration entry of the given nameValue stored in the given system collection.
+    * Returns an DataDocument containing all configurations for unique nameValue.
     *
     * @param collectionName
     *       the name of collection in system database
-    * @param nameValue
+    * @param identificator
     *       the unique name value of stored configuration entry
-    * @return configuration entry of the given nameValue in given collection
+    * @return List storing all configurations
     */
-   public Optional<DataDocument> getConfigurationEntry(final String collectionName, final String nameValue) {
-      if (nameValue == null || !systemDataStorage.hasCollection(collectionName)) {
-         return Optional.empty();
-      }
+   public List<Config> getConfigurations(final String collectionName, final String identificator) {
+      DataDocument document = systemDataStorage.readDocument(collectionName, entryFilter(identificator));
+      List<DataDocument> configs = document != null ? document.getArrayList(LumeerConst.Configuration.CONFIGS, DataDocument.class) : Collections.emptyList();
 
-      final DataFilter filter = dataStorageDialect.fieldValueFilter(NAME_KEY, nameValue);
-      DataDocument config = systemDataStorage.readDocument(collectionName, filter);
-      return Optional.ofNullable(config);
+      return configs.stream().map(Config::new).collect(Collectors.toList());
    }
 
-   public boolean hasConfigurationAttributeFlag(final String collectionName, final String configName, final String attributeName, final String flagName) {
-      Optional<DataDocument> configuration = getConfigurationEntry(collectionName, configName);
-      if (!configuration.isPresent()) {
-         return false;
-      }
-
-      List<String> flags = configuration.get().getArrayList(FLAGS_KEY, String.class);
-      if (flags.isEmpty()) {
+   public boolean hasConfigurationAttributeFlag(final String collectionName, final String identificator, final String attributeName, final String flagName) {
```

Comment:
I think we do not need this flag mechanism in `ConfigurationManipulator` anymore since the flags can be obtained directly from configuration entries.